### PR TITLE
Simplify Windows platform identification in Gemfile

### PR DESCRIPTION
**Pull request by bf4**
Platforms map available [here](https://github.com/bundler/bundler/blob/89a8778c19269561926cea172acdcda241d26d23/lib/bundler/dependency.rb#L30-L54).

### CODE REVIEW

**Comment by beauby** [comment]
#1120 mentions something about that.

**Comment by bf4** [comment]
Included #1120 but not #1126

File: Gemfile [hunk 1]
```diff
@@ -25,9 +25,14 @@ else
   gem 'actionpack', gem_version
 end
 
+@windows_platforms = [:mingw, :x64_mingw, :mswin]
```
Comment (joaomdmoura) [inline Gemfile]: :+1:

**Comment by joaomdmoura** [comment]
Let's wait for @Eric-Guo comments on #1120 before moving on with this

File: Gemfile (hunk 1 above)
Comment (bf4) [inline Gemfile]: hmmm https://github.com/bundler/bundler/search?utf8=%E2%9C%93&q=mswin

https://github.com/bundler/bundler/blob/1b3eb2465a64d938490110e84b5ca97a51253d14/lib/bundler/constants.rb#L2

>   WINDOWS = RbConfig::CONFIG["host_os"] =~ /(msdos|mswin|djgpp|mingw)/

> https://github.com/bundler/bundler/blob/89a8778c19269561926cea172acdcda241d26d23/lib/bundler/current_ruby.rb#L118-L216
> https://github.com/bundler/bundler/blob/89a8778c19269561926cea172acdcda241d26d23/lib/bundler/dependency.rb#L30-L54

**Comment by Eric-Guo** [comment]
Coverage still high after merge #1120, should be appveyor problem

ams_coverage

**Comment by bf4** [comment]
Hmm, windows coverage seems fine in https://ci.appveyor.com/project/joaomdmoura/active-model-serializers/build/38

File: .simplecov [hunk 2]
```diff
@@ -3,10 +3,13 @@
 # vim: set ft=ruby
 
 ## DEFINE VARIABLES
+require 'rbconfig'
 @minimum_coverage = ENV.fetch('COVERAGE_MINIMUM') {
   case (defined?(RUBY_ENGINE) && RUBY_ENGINE) || "ruby"
   when 'jruby', 'rbx'
     96.0
+  when !!(RbConfig::CONFIG['host_os'] =~ /cygwin|mswin|mingw|bccwin|wince|emx/) # windows
+    50.0
```
Comment (NullVoxPopuli) [inline .simplecov]: why 50%?

**Comment by bf4** [comment]
Might be nice to have, but not terribly important

File: Gemfile [hunk 3]
```diff
@@ -29,20 +29,25 @@ else
...
-  gem 'sqlite3', platform: [:ruby, :mingw, :x64_mingw, :mswin]
+  gem 'sqlite3',                          platform: (@windows_platforms + [:ruby])
   gem 'activerecord-jdbcsqlite3-adapter', platform: :jruby
-  gem 'codeclimate-test-reporter', require: false
-end
 
-group :test, :development do
-  gem 'simplecov', '~> 0.10', require: false
+  gem 'codeclimate-test-reporter', require: false
+  gem 'simplecov', '~> 0.10', require: false, group: :development
```
Comment (NullVoxPopuli) [inline Gemfile]: does this allow simplecov in both test and dev?

File: Gemfile [hunk 4]
```diff
@@ -29,20 +29,25 @@ else
   gem 'actionpack', gem_version
 end
 
+# https://github.com/bundler/bundler/blob/89a8778c19269561926cea172acdcda241d26d23/lib/bundler/dependency.rb#L30-L54
+# https://github.com/bundler/bundler/blob/89a8778c19269561926cea172acdcda241d26d23/lib/bundler/current_ruby.rb#L118-L216
+# https://github.com/bundler/bundler/blob/1b3eb2465a64d938490110e84b5ca97a51253d14/lib/bundler/constants.rb#L2
+# require 'rbconfig'
+# !!(RbConfig::CONFIG['host_os'] =~ /cygwin|mswin|mingw|bccwin|wince|emx/) # windows
+@windows_platforms = [:mswin, :mingw, :x64_mingw]
```
Comment (beauby) [inline Gemfile]: According to https://github.com/bundler/bundler/blob/89a8778c19269561926cea172acdcda241d26d23/lib/bundler/current_ruby.rb#L118-L216, we could get away with simply `:mswin`, or have I missed something?

Comment (bf4) [inline Gemfile]: I feel like I tried that and it wasn't sufficient, but YMMV

**Comment by beauby** [comment]
So it seems `mswin64` platform was added only in bundler 1.8. I'll remove it for now, but let's make a sticky note somewhere that someday we'll have to add it.

**Pull request merged**